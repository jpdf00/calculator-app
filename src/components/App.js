import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const data = calculate(this.state, buttonName);
    this.setState(data);
  }

  render() {
    /* eslint-disable no-unused-vars */
    const { total, next, operation } = this.state;
    /* eslint-enable no-unused-vars */
    const result = !next ? total : next;

    return (
      <>
        <Display result={result || '0'} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
