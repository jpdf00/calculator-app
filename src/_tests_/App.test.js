import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../components/App';

describe('App Component', () => {
  const btns = ['1', '2', '3', '4', '5', '6', '7', '8', '9', 'AC', '+/-', '.', '+', '-', 'ร', 'รท', '%', '='];

  test('It should render', () => {
    const app = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });

  btns.forEach((item) => {
    test(`should render the button ${item}`, () => {
      const { getByText } = render(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      );
      const btnName = getByText(item);
      expect(btnName).toBeInTheDocument();
    });
  });
});
