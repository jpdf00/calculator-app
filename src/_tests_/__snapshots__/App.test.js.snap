// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component It should render 1`] = `
<div
  className="Calculator"
>
  <h2>
    Let's do some Math!
  </h2>
  <div
    className="Size App"
  >
    <h2
      className="Size Display"
    >
      0
    </h2>
    <div
      className="Size ButtonPanel"
    >
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5913e",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="Size ButtonPanel"
    >
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5913e",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        ร
      </button>
    </div>
    <div
      className="Size ButtonPanel"
    >
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5913e",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="Size ButtonPanel"
    >
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5913e",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="Size ButtonPanel"
    >
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "undefined",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        name="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5913e",
            "border": "1px solid #d8d8d8",
            "fontSize": "24px",
            "width": "25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
